### PR TITLE
Updated version and changelog.md towards v0.7.0

### PR DESCRIPTION
**Pull request by jvdp1**
Updated `VERSION` and `CHANGELOG.md` towards the release v0.7.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,43 @@
+# Version 0.7.0
+
+Full release notes available at [v0.7.0] tag.
+
+[v0.7.0]: https://github.com/fortran-lang/stdlib/releases/tag/v0.7.0
+
+- new module `stdlib_constants`
+  [#800](https://github.com/fortran-lang/stdlib/pull/800)
+  - Many mathematical constants and most common physical (codata) constants
```
Comment (perazz) [inline CHANGELOG.md:9]:
```suggestion
  - Many mathematical constants and most common physical ([codata](https://codata.org)) constants
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+  - changes in module `stdlib_linalg`
+    - Support of extended and quad precision checking 
+      [#821](https://github.com/fortran-lang/stdlib/pull/821)
+    - Several fixes
+      [#815](https://github.com/fortran-lang/stdlib/pull/815)
+      [#818](https://github.com/fortran-lang/stdlib/pull/818)
+      [#826](https://github.com/fortran-lang/stdlib/pull/826)
+      [#830](https://github.com/fortran-lang/stdlib/pull/830)
+      [#836](https://github.com/fortran-lang/stdlib/pull/836)
+    - New procedures for Singular Value Decomposition: `eig`, `eigh`, `eigvals`, `eigvalsh`
```
Comment (perazz) [inline CHANGELOG.md:30]:
```suggestion
    - New procedures for Eigenvalues and Eigenvectors computation: `eig`, `eigh`, `eigvals`, `eigvalsh`
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+      [#808](https://github.com/fortran-lang/stdlib/pull/808)
+    - New procedures for Eigenvalues and Eigenvectors computation: `svd`, `svdvals`
```
Comment (perazz) [inline CHANGELOG.md:32]:
```suggestion
    - New procedures for Singular Value Decomposition: `svd`, `svdvals`
```

**Review by perazz — APPROVED** [review]
LGTM, thank you @jvdp1

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+    - Support of extended and quad precision checking 
+      [#821](https://github.com/fortran-lang/stdlib/pull/821)
+    - Several fixes
+      [#815](https://github.com/fortran-lang/stdlib/pull/815)
+      [#818](https://github.com/fortran-lang/stdlib/pull/818)
+      [#826](https://github.com/fortran-lang/stdlib/pull/826)
+      [#830](https://github.com/fortran-lang/stdlib/pull/830)
+      [#836](https://github.com/fortran-lang/stdlib/pull/836)
+    - New procedures for Singular Value Decomposition: `eig`, `eigh`, `eigvals`, `eigvalsh`
+      [#808](https://github.com/fortran-lang/stdlib/pull/808)
```
Comment (jvdp1) [inline CHANGELOG.md:31]:
```suggestion
      [#816](https://github.com/fortran-lang/stdlib/pull/816)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,43 @@
...
+    - New procedures for Eigenvalues and Eigenvectors computation: `svd`, `svdvals`
+      [#816](https://github.com/fortran-lang/stdlib/pull/816)
```
Comment (jvdp1) [inline CHANGELOG.md:33]:
```suggestion
      [#808](https://github.com/fortran-lang/stdlib/pull/808)
```

**Comment by jvdp1** [comment]
Thank you @perazz for the fixes and review. I will merge it and generate the release

**Pull request merged**